pinot-common/src/main/java/org/apache/pinot/common/function/scalar/comparison/NotEqualsScalarFunction.java

```diff
@@ -0,0 +1,154 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.common.function.scalar.comparison;
+
+import java.math.BigDecimal;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import org.apache.pinot.common.function.FunctionInfo;
+import org.apache.pinot.common.utils.DataSchema;
+import org.apache.pinot.spi.annotations.ScalarFunction;
+
+/**
+ * Polymorphic notEquals (!=) scalar function implementation
+ */
+@ScalarFunction
+public class NotEqualsScalarFunction extends PolymorphicComparisonScalarFunction {
+
+  private static final Map<DataSchema.ColumnDataType, FunctionInfo> TYPE_FUNCTION_INFO_MAP = new HashMap<>();
+
+  static {
+    try {
+      // Set nullable parameters to false for each function because the return value should be null if any argument
+      // is null
+      TYPE_FUNCTION_INFO_MAP.put(DataSchema.ColumnDataType.INT, new FunctionInfo(
+          NotEqualsScalarFunction.class.getMethod("intNotEquals", int.class, int.class),
```

Comment:
Is there a way to directly access a method without reflection?

Reply:
Hmm, the `FunctionInfo` class is using `java.lang.reflect.Method` to store the method reference and this is invoked directly in `FunctionInvoker` via `Method::invoke`, so I don't think it's possible to avoid using reflection here. Although since this is in a static initializer block that will only be executed once in a JVM when each of these classes is initialized, it shouldn't be much of a performance concern right?

Reply:
Yes, performance is not a concern. I also didn't find a better way, but I do want to see if it is possible to improve readability by directly referencing the method :-P